spec/services/pagination_service_spec.rb

```diff
@@ -0,0 +1,689 @@
+            else
+              limit_error = response['errors'].first
+              offset_error = response['errors'].second
+            end
+
+            expect(offset_error['status']).to eq '200'
+            expect(offset_error['source']).to include("page_offset" => "40")
+            expect(offset_error['title']).to eq 'Page Offset Out of Range'
+            expect(offset_error['detail']).to eq "Page offset 40 > 36 (total number of results).  Returning empty results."
+
+            expect(limit_error['status']).to eq '200'
+            expect(limit_error['source']).to include("page_limit" => "0")
+            expect(limit_error['title']).to eq 'Page Limit Out of Range'
+            expect(limit_error['detail']).to eq "Page limit 0 < 1 (minimum limit).  Returning empty results."
+          end
+
+          it 'returns json api response with no results' do
+            expect(response["data"]).to match_array([])
+          end
+
+          it "sets meta['page'] stats with page_offset and page_limit showing the passed in values" do
+            expect(response["meta"]["page"]["page_offset"]).to eq "40"
+            expect(response["meta"]["page"]["page_limit"]).to eq "0"
+            expect(response["meta"]["page"]["actual_page_size"]).to eq "0"
+            expect(response["meta"]["page"]["total_num_found"]).to eq "36"
+          end
+
+          it "sets prev and next links to nil" do
+            expect(response['links']['prev']).to be_nil
+            expect(response['links']['next']).to be_nil
+          end
+
+          it "uses request values in self link" do
+            self_url = "#{base_url}#{url_path}?q=term&format=jsonapi&page_limit=0&page_offset=40"
+            expect(response['links']['self']).to eq self_url
+          end
+
+          it "uses valid values in first and last links" do
+            expect(response['links']['first']).to eq first_page
+            expect(response['links']['last']).to eq last_page
+          end
+        end
+      end
+
+      context 'with page_offset missing' do
+        context 'and page_limit is missing' do
+          it 'returns the first DEFAULT_PAGE_LIMIT (10) records' do
+            expect(response["data"]).to match_array(results[0..9])
+          end
+
+          it "sets meta['page'] stats" do
+            expect(response["meta"]["page"]["page_offset"]).to eq "1"
+            expect(response["meta"]["page"]["page_limit"]).to eq described_class::DEFAULT_PAGE_LIMIT.to_s
+            expect(response["meta"]["page"]["actual_page_size"]).to eq described_class::DEFAULT_PAGE_LIMIT.to_s
+            expect(response["meta"]["page"]["total_num_found"]).to eq "36"
+          end
+
+          it "sets prev link to nil and next link to next page" do
+            next_url = "#{base_url}#{url_path}?q=term&format=jsonapi&page_limit=10&page_offset=11"
+            expect(response['links']['prev']).to be_nil
+            expect(response['links']['next']).to eq next_url
```

Comment:
I think you can use `second_page` here:
```suggestion
            expect(response['links']['prev']).to be_nil
            expect(response['links']['next']).to eq second_page
```